cget/util.py

```diff
@@ -87,6 +87,43 @@ def mkfile(d, file, content, always_write=True):
         write_to(p, content)
     return p
 
+def cache_lock():
+    cache_base_dir = get_cache_path()
+    mkdir(cache_base_dir)
+    return filelock.FileLock(os.path.join(cache_base_dir, "lock"))
+
+def zipdir(src_dir, tgt_file):
+    print("zipping '%s' to '%s" % (src_dir, tgt_file))
+    zipf = zipfile.ZipFile(tgt_file, 'w', zipfile.ZIP_DEFLATED)
```

Comment:
Is is possible to use a tar file instead? This way we can pick different compression schemes(ie .gz vs .bzip2 vs .xz).

Reply:
I discovered archiving is actually wrong. The packages config files reference the install path, so it needs to be constant. The right approach is to do what conan does and install into the cache. If you want remote caching you need to make packages relocatable, which conan also does, but is a bit more tricky i think

Reply:
> The packages config files reference the install path, so it needs to be constant.

The export files generated by cmake should be relocatable as long as you dont add hard-coded paths to dependencies(and instead use import targets and `find_dependency`). Pkg-config files usually have a hard-coded path like `prefix=<install-prefix>`, but pkg-config will adjust this prefix path if the packages are moved.

> If you want remote caching you need to make packages relocatable, which conan also does, but is a bit more tricky i think

Ideally, a library component(and not a package manager) should make their installation relocatable, but this requires upstreaming usage requirements for all libraries(which will benefit the entire eco-system rather than just one package manager).

Reply:
Well, yeah, but it seems to me much cheaper to just install to the cache. The only downside is losing remote caching, so i tend towards it right now, and solve it later for remote caching?

Reply:
Yea that should work for local caching.

Reply:
just checked: ffmpegs pkgconfig file is not relocatabale already